backend/kernelCI_app/views/treeCommitsHistory.py

```diff
@@ -54,8 +54,9 @@ def __treat_unknown_filter(self, table_field, op, value_name, filter):
         clause = table_field
         is_null_clause = f"{table_field} IS NULL"
         has_null_value = False
-        if (isinstance(filter['value'], str)):
-            filter['value'] = '' if filter['value'] in ['NULL', 'Unknown'] else filter['value']
+
+        if (isinstance(filter['value'], str) and filter['value'] in ['NULL', 'Unknown']):
```

Comment:
```suggestion
        if (filter['value'] is str and filter['value'] in ['NULL', 'Unknown']):
```

Reply:
These are actually not equivalent, `is str` would compare if `filter['value']` is literally `str`, while `isinstance` would check if the class of `filter['value']` is str